### PR TITLE
Update links to point to a feature branch

### PR DESCRIPTION
**Pull request by antkmsft**

### CODE REVIEW

File: sdk/samples/storage/blobs/README.md [hunk 1]
```diff
@@ -52,11 +52,11 @@ Azure SDK for Embedded C is licensed under the [MIT][azure_sdk_for_c_license] li
 <!-- LINKS -->
 [azure_sdk_for_c_contributing]: https://github.com/Azure/azure-sdk-for-c/blob/main/CONTRIBUTING.md
 [azure_sdk_for_c_license]: https://github.com/Azure/azure-sdk-for-c/blob/main/LICENSE
-[BLOB_SDK_README]: https://github.com/Azure/azure-sdk-for-c/tree/main/sdk/docs/storage
-[SDK_README_CONTRIBUTING]:https://github.com/Azure/azure-sdk-for-c/tree/main/sdk/docs/storage#contributing
-[SDK_README_GETTING_STARTED]: https://github.com/Azure/azure-sdk-for-c/tree/main/sdk/docs/storage#getting-started
-[SDK_README_KEY_CONCEPTS]: https://github.com/Azure/azure-sdk-for-c/tree/main/sdk/docs/storage#key-concepts
-[samples_basic]: https://github.com/Azure/azure-sdk-for-c/blob/main/sdk/samples/storage/blobs/src/blobs_client_sample.c
+[BLOB_SDK_README]: https://github.com/Azure/azure-sdk-for-c/tree/feature/storage-blobs/sdk/docs/storage
+[SDK_README_CONTRIBUTING]:https://github.com/Azure/azure-sdk-for-c/tree/feature/storage-blobs/sdk/docs/storage#contributing
```
Comment (ahsonkhan) [inline sdk/samples/storage/blobs/README.md:56]: nit:
```suggestion
[SDK_README_CONTRIBUTING]: https://github.com/Azure/azure-sdk-for-c/tree/feature/storage-blobs/sdk/docs/storage#contributing
```

**Review by ahsonkhan — APPROVED** [review]

**Comment by ahsonkhan** [comment]
Follow-up to resolve release issue:
https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1187023&view=logs&j=a129effc-2dd1-54d1-fb5a-ad7bdc0e851d&t=617ceecd-bf48-5ce3-83e2-7468b1632a13

See https://github.com/Azure/azure-sdk-for-c/pull/1997#issuecomment-964647839

File: sdk/docs/storage/README.md [hunk 2]
```diff
@@ -99,7 +99,7 @@ Azure SDK for Embedded C is licensed under the [MIT][azure_sdk_for_c_license] li
 [azure_sdk_for_c_license]: https://github.com/Azure/azure-sdk-for-c/blob/main/LICENSE
 [azure_sdk_for_c_readme]: https://github.com/Azure/azure-sdk-for-c/blob/main/README.md
 [azure_sdk_for_c_readme_getting_started]:https://github.com/Azure/azure-sdk-for-c/blob/main/README.md#getting-started-using-the-sdk
```
Comment (antkmsft) [inline sdk/docs/storage/README.md:101]:
```suggestion
[azure_sdk_for_c_readme_getting_started]: https://github.com/Azure/azure-sdk-for-c/blob/main/README.md#getting-started-using-the-sdk
```

File: sdk/samples/storage/blobs/README.md [hunk 3]
```diff
@@ -52,11 +52,11 @@ Azure SDK for Embedded C is licensed under the [MIT][azure_sdk_for_c_license] li
 <!-- LINKS -->
 [azure_sdk_for_c_contributing]: https://github.com/Azure/azure-sdk-for-c/blob/main/CONTRIBUTING.md
 [azure_sdk_for_c_license]: https://github.com/Azure/azure-sdk-for-c/blob/main/LICENSE
-[BLOB_SDK_README]: https://github.com/Azure/azure-sdk-for-c/tree/main/sdk/docs/storage
-[SDK_README_CONTRIBUTING]:https://github.com/Azure/azure-sdk-for-c/tree/main/sdk/docs/storage#contributing
-[SDK_README_GETTING_STARTED]: https://github.com/Azure/azure-sdk-for-c/tree/main/sdk/docs/storage#getting-started
-[SDK_README_KEY_CONCEPTS]: https://github.com/Azure/azure-sdk-for-c/tree/main/sdk/docs/storage#key-concepts
-[samples_basic]: https://github.com/Azure/azure-sdk-for-c/blob/main/sdk/samples/storage/blobs/src/blobs_client_sample.c
```
Comment (ahsonkhan) [inline sdk/samples/storage/blobs/README.md:59]: Interesting that the failed leg only detected two invalid links, when there were some more:
https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1187023&view=logs&j=a129effc-2dd1-54d1-fb5a-ad7bdc0e851d&t=617ceecd-bf48-5ce3-83e2-7468b1632a13
> 'file:///D:/a/_work/1/s/sdk/samples/storage/blobs/README.md' has 2 broken link(s):
  https://github.com/Azure/azure-sdk-for-c/tree/main/sdk/docs/storage
  https://github.com/Azure/azure-sdk-for-c/blob/main/sdk/samples/storage/blobs/src/blobs_client_sample.c

File: sdk/samples/storage/blobs/README.md [hunk 4]
```diff
@@ -52,11 +52,11 @@ Azure SDK for Embedded C is licensed under the [MIT][azure_sdk_for_c_license] li
 <!-- LINKS -->
 [azure_sdk_for_c_contributing]: https://github.com/Azure/azure-sdk-for-c/blob/main/CONTRIBUTING.md
 [azure_sdk_for_c_license]: https://github.com/Azure/azure-sdk-for-c/blob/main/LICENSE
-[BLOB_SDK_README]: https://github.com/Azure/azure-sdk-for-c/tree/main/sdk/docs/storage
-[SDK_README_CONTRIBUTING]:https://github.com/Azure/azure-sdk-for-c/tree/main/sdk/docs/storage#contributing
```
Comment (ahsonkhan) [inline sdk/samples/storage/blobs/README.md:56]: Btw, this link is not being referenced anywhere in the markdown and can be remove (unrelated cleanup, we can do later)

**Pull request merged**